### PR TITLE
added generic post and get functionality with notes

### PR DESCRIPTION
**Pull request by SeanRWelsh**
Now have the ability to create a note with questions via a post reqeust to /notes with json as the body ex:
```
{
    "content":
      "Addition is the process of adding two or more items together. Addition in Maths is the method of calculating the sum of two or more numbers. It    is a primary arithmetic operation that is used commonly in our day-to-day life. One of the most common uses of addition is when we work    with money, calculate our grocery bills, or calculate time. In this    article, let us learn more about the addition definition, the addition    symbol, addition sums, the parts of addition, addition with regrouping,    and number line addition, along with some addition examples.",

    "title": "Addition",
    "summary": "Addition is the process of adding two or more items together.",
    "questions": [
      "What is 2+2?",
      "Compare and contrast addition and subtraction.",
      "What is the difference between addition and multiplication?",
      "What is 67 + 12391?"
    ]
  }
```
will create

```
[
    {
        "idNote": 2,
        "title": "Addition",
        "content": "Addition is the process of adding two or more items together. Addition in Maths is the method of calculating the sum of two or more numbers. It    is a primary arithmetic operation that is used commonly in our day-to-day life. One of the most common uses of addition is when we work    with money, calculate our grocery bills, or calculate time. In this    article, let us learn more about the addition definition, the addition    symbol, addition sums, the parts of addition, addition with regrouping,    and number line addition, along with some addition examples.",
        "dateCreated": "2024-02-02T20:19:57.943+00:00",
        "dateModified": null,
        "summary": "Addition is the process of adding two or more items together.",
        "questions": [
            {
                "idQuestion": 7,
                "content": "Compare and contrast addition and subtraction."
            },
            {
                "idQuestion": 8,
                "content": "What is 2+2?"
            },
            {
                "idQuestion": 6,
                "content": "What is 67 + 12391?"
            },
            {
                "idQuestion": 5,
                "content": "What is the difference between addition and multiplication?"
            }
        ]
    }
]
```
can also fetch all note with a GET request to /notes
once user functionality will update to be user specific. Also what stipulations if any do we want to put on the data so that we can start handling validation and errors

### CODE REVIEW

File: src/main/java/com/cornellnotes/Cornell/Notes/Backend/models/Question.java
Comment (krishanleong) [inline src/main/java/com/cornellnotes/Cornell/Notes/Backend/models/Question.java:1]: Should we declare a content variable in this class?

**Review by krishanleong — COMMENTED** [review]
That's the only thing I see.

File: src/main/java/com/cornellnotes/Cornell/Notes/Backend/models/Question.java
Comment (SeanRWelsh) [reply src/main/java/com/cornellnotes/Cornell/Notes/Backend/models/Question.java:1]: There is a string content variable on line 25 or are you talking about something else ?

Comment (krishanleong) [reply src/main/java/com/cornellnotes/Cornell/Notes/Backend/models/Question.java:1]: No I missed it.  I didn't realize I wasn't looking a the whole file.

**Pull request merged**